Poet engine fixes

aludvik opened the pull request.

Commit:
Move publishing activity into separate method

This keeps the level of abstraction in the main loop consistent.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Queue forks while waiting for block to commit

After the PoET engine has decided to commit a new fork, it will queue up valid
blocks that need to have fork resolution done on them until the last fork
resolution decision has been committed.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Remove unneeded chain_head request

The chain head is the newly committed block.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Catch and log unhandled consensus driver exception

Log any unhandled exceptions from the python consensus sdk driver.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Wait before trying again on summarize empty block

The PoET engine needs to wait a second before re-requesting the validator to
summarize a block to give it a chance to do some work.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Fix Python CSDK driver bug

There was a bug in the Python CSDK driver that would cause messages to be lost
if the future timed out. A new future would be created but the message would be
resolved in the old future. Instead of creating a new future, hold onto the
old future and poll it until it completes.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Parse finalize block args before printing

Improve log message readibility by parsing the finalize block results before
logging them.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Change log level of not ready messages

These messages will occur normally if there are no batches being sent to the network.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

boydjohnson commented on integration/sawtooth_integration/docker/test_poet_engine.yaml:
```diff
@@ -0,0 +1,136 @@
+    command: tail -f /dev/null
+
+  intkey-workload:
+    image: integration-tests:$ISOLATION_ID
```
This should use a `build:` section for building the image.

Commit:
Cleanup PoET engine logging message

Remove redundant logs and add in a log to show when a validator publishes, what
block it published.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>
